Doc/c-api/time.rst

```diff
@@ -0,0 +1,90 @@
+.. highlight:: c
+
+PyTime C API
+============
+
+.. versionadded:: 3.13
+
+PyTime API
+----------
+
+The PyTime API provides access to system clocks and time conversion functions.
+It is similar to the Python :mod:`time` module.
+
+
+Types
+-----
+
+.. c:type:: PyTime_t
+
+   A timestamp or duration in nanoseconds represented as a 64-bit signed
+   integer.
```

Comment:
At the C-API WG issue, you say that the units are supposed to be an implementation detail.

Reply:
Oh, now I'm confused. I don't recall the exact history of this PR that I wrote 3 months ago.

When I wrote that API 10 years ago, I tried hard to not expose the "unit" (1 nanosecond). In practice, it was always 1 nanosecond, even if on Windows, clocks only have a resolution of 100 nanoseconds in the best case.

Maybe it's time to stick to nanoseconds. On Linux, `timespec`, `clock_gettime()`, `nanosleep()` have a resolution of 1 nanosecond. I don't know any operating system API using a better resolution.

So well, we should stick to 1 nanosecond to avoid any confusion. So creating a `PyTime_t` of 1 nanosecond is just `PyTime_t one_ns = 1;`. No "from nanoseconds" or "to nanoseconds" are needed. It makes the API simpler to use (so less error prone).